### PR TITLE
PP-6362: Add screen height to parameters

### PR DESCRIPTION
**Pull request by oswaldquek**
Provide screen height.

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java [hunk 1]
```diff
@@ -30,11 +36,21 @@ public EpdqPayloadDefinitionForNew3ds2Order(String frontendUrl, SupportedLanguag
...
-                .add(BROWSER_LANGUAGE, getBrowserLanguage(templateData));
+                .add(BROWSER_LANGUAGE, getBrowserLanguage(templateData))
+                .add(BROWSER_SCREEN_HEIGHT, getBrowserScreenHeight(templateData));
 
         return parameterBuilder.build();
     }
 
+    private String getBrowserScreenHeight(EpdqTemplateData templateData) {
+        var screenHeight = templateData.getAuthCardDetails().getJsScreenHeight().orElse(DEFAULT_BROWSER_SCREEN_HEIGHT);
+        if (isCreatable(screenHeight) && BROWSER_SCREEN_HEIGHT_RANGE.contains(parseInt(screenHeight))) {
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java]: I think `isCreatable(…)` will allow numbers that can’t be parsed into integers (like floats).

I think a regex is possibly the right approach here. `0|[1-9][0-9]{0,5}` should work (either just 0, _or_ a digit from 1 to 9 followed by up to five digits from 0 to 9, thus allowing numbers from 0 to 999999 without allowing leading zeroes, which would get parsed as octal and give unexpected results).

I think this method could avoid the need for Apache Commons altogether if it was something like this (assume `NUMBER_FROM_0_TO_999999` is a constant `Pattern` of the regex above and note the change of return type to `int`):

```java
private int getBrowserScreenHeight(EpdqTemplateData templateData) {
    return templateData.getAuthCardDetails().getJsScreenHeight()
            .filter(screenHeight -> NUMBER_FROM_0_TO_999999.matcher(screenHeight).matches())
            .map(Integer::parseInt)
            .filter(screenHeight -> screenHeight >= 0 && screenHeight <= 999999)
            .orElse(DEFAULT_BROWSER_SCREEN_HEIGHT);
}
```

File: src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java [hunk 2]
```diff
@@ -86,6 +88,22 @@ public void setup() {
...
     
+    @Test
+    public void should_include_browserScreenHeight() {
+        authCardDetails.setJsScreenHeight("100");
+        List<NameValuePair> result = epdqPayloadDefinitionFor3ds2NewOrder.extract(epdqTemplateData);
+        assertThat(result, is(aParameterBuilder().withBrowserScreenHeight("100").build()));
+    }
+    
+    @Test
+    @Parameters({"null", "-1", "1000000", "invalid"})
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java]: Maybe toss some non-integers into here and some hex or something

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks good. A couple of suggestions.

**Comment by oswaldquek** [comment]
@alexbishop1 Sharp. I've addressed the comments.

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java [hunk 3]
```diff
@@ -30,11 +34,21 @@ public EpdqPayloadDefinitionForNew3ds2Order(String frontendUrl, SupportedLanguag
...
 
         return parameterBuilder.build();
     }
 
+    private String getBrowserScreenHeight(EpdqTemplateData templateData) {
+        return templateData.getAuthCardDetails().getJsScreenHeight()
+                .filter(screenHeight -> NUMBER_FROM_0_TO_999999.matcher(screenHeight).matches())
+                .map(Integer::parseInt)
+                .filter(screenHeight -> screenHeight >= 0 && screenHeight <= 999999)
+                .map(screenHeight -> Integer.toString(screenHeight))
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java:48]: Just `map(Integer::toString)` would work here

File: src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java [hunk 4]
```diff
@@ -86,6 +88,23 @@ public void setup() {
...
+    @Test
+    @Parameters({"0", "999999", "100"})
+    public void should_include_browserScreenHeight(String screenHeight) {
+        authCardDetails.setJsScreenHeight(screenHeight);
+        List<NameValuePair> result = epdqPayloadDefinitionFor3ds2NewOrder.extract(epdqTemplateData);
+        assertThat(result, is(aParameterBuilder().withBrowserScreenHeight(screenHeight).build()));
+    }
+    
+    @Test
+    @Parameters({"null", "-1", "1000000", "invalid", "0x6", "123L", "1.2"})
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java:100]: Parameterised tests really shine for this sort of stuff

**Review by alexbishop1 — APPROVED** [review]

File: src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java (hunk 3 above)
Comment (oswaldquek) [reply src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java:48]: it doesn't!

Comment (alexbishop1) [reply src/main/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2Order.java:48]: Of course! It’s ambiguous whether it’s the `toString()` instance method or the static `toString(int)` method!

**Pull request merged**